internal/controller/appserver/reconciler.go

```diff
@@ -343,18 +343,35 @@ func reconcileDeployment(r reconciler.Reconciler, ctx context.Context, cr *olsv1
 	existingDeployment := &appsv1.Deployment{}
 	err = r.Get(ctx, client.ObjectKey{Name: utils.OLSAppServerDeploymentName, Namespace: r.GetNamespace()}, existingDeployment)
 	if err != nil && errors.IsNotFound(err) {
-		utils.UpdateDeploymentAnnotations(desiredDeployment, map[string]string{
-			utils.OLSConfigHashKey:      r.GetStateCache()[utils.OLSConfigHashStateCacheKey],
-			utils.OLSAppTLSHashKey:      r.GetStateCache()[utils.OLSAppTLSHashStateCacheKey],
-			utils.LLMProviderHashKey:    r.GetStateCache()[utils.LLMProviderHashStateCacheKey],
-			utils.PostgresSecretHashKey: r.GetStateCache()[utils.PostgresSecretHashStateCacheKey],
-		})
-		utils.UpdateDeploymentTemplateAnnotations(desiredDeployment, map[string]string{
+		// Before creating app server deployment, verify PostgreSQL is ready (if using postgres cache)
+		if cr.Spec.OLSConfig.ConversationCache.Type == utils.OLSDefaultCacheType {
+			postgresDeployment := &appsv1.Deployment{}
+			err := r.Get(ctx, client.ObjectKey{Name: utils.PostgresDeploymentName, Namespace: r.GetNamespace()}, postgresDeployment)
+			if err == nil {
+				// PostgreSQL deployment exists, check if it's ready
+				if postgresDeployment.Status.ReadyReplicas != *postgresDeployment.Spec.Replicas {
+					// PostgreSQL is not ready yet, skip app-server creation to avoid readiness failures
+					r.GetLogger().Info("PostgreSQL deployment is not ready yet, deferring app-server creation until PostgreSQL is ready",
+						"postgres_ready_replicas", postgresDeployment.Status.ReadyReplicas,
+						"postgres_desired_replicas", *postgresDeployment.Spec.Replicas)
+					return nil
```

Comment:
Should this be returning  nil - what are we expecting to do here ? Should we not reque in this case ?